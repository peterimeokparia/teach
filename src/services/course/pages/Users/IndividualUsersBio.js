import React, { 
useState } from 'react';

import { 
connect } from 'react-redux';

import { 
navigate } from '@reach/router';

import {  
getOperatorFromOperatorBusinessName } from '../../Selectors';


import { 
forceReload } from '../../../../helpers/serverHelper';


import LoginLogout from '../Login/LoginLogout'
import Cart from '../Sales/Cart/Cart';
import UserBioEditor from './UserBioEditor';
import ImageComponent from '../Components/ImageComponent';

import './MyCourses.css';



const IndividualUsersBio = ({ user, users, userId, operatorBusinessName, operator }) => {

  const [ editing, setEditing ] = useState( false );


  if ( ! user?.userIsValidated || ! operator ){

      navigate(`/${operatorBusinessName}/login`);
  }



  const toggleEditing = () => {

       if ( editing ) {

         setEditing( false );

         forceReload();

       } else {

         setEditing( true );

       }

  }


  let tutor = Object.values(users)?.find(user => user?._id === userId);

  const fileUploadUrl = 'http://localhost:9005/api/v1/fileUploads/avatar';

  return  (

               <div> 
                        <div className="MyCourses">

                                <header> 
                                    <h1>  {`Welcome ${user?.firstname}! `} </h1>

                                    {/* <h2> You are viewing {tutor?.firstname}'s Bio. </h2> */}

                                        <div>  

                                        <LoginLogout
                                              operatorBusinessName
                                              user 
                                        />

                                        <Cart />

                                        </div>
                                    </header>

                                    <br></br>

                                     { (user?.role === "Tutor") && <button className="view-bio-btn" onClick={toggleEditing}>{editing ? 'Preview' : 'Edit'}</button> }
                            
                                     <h1>  {`${tutor?.firstname} `} </h1>
               
                                <div className="bio-image-btn">

                                        <ImageComponent
                                              user={tutor}
                                              url={fileUploadUrl} 
                                              editMode={editing}
                                              typeOfUpload={'useravatarbiourl'}
                                              imageSrc={tutor?.avatarUrl}
                                              teachObjectName={"users"}
                                        />

                                </div> 

                                                <div className="content"> 
                                                      
                                                <div className="sidebar" />

                                                   <div className={"bio-editor"}>
                                                   <UserBioEditor user={user}/>
                                                   </div> 
                                                     
                                                   

                                                </div> 
                                       
                                    <br></br>
                                 
                       </div>
                        
                    </div>
          
                );
            }



const mapState = (state, ownProps)   => {
  return {
         operator: getOperatorFromOperatorBusinessName(state, ownProps),
         previewMode: state.app.previewMode,
         users: state.users.users,
         user:state.users.user
  };
}


export default connect( mapState )(IndividualUsersBio);
