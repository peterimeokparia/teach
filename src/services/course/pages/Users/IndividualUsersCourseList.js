import React from 'react';

import { 
connect } from 'react-redux';

import { 
Link,
navigate } from '@reach/router';

import {  
getUsersByOperatorId,
getCoursesByOperatorId } from '../../Selectors';

import Loading from '../Components/Loading';
import LoginLogout from '../Login/LoginLogout'
import CoursesComponent from '../Courses/CoursesComponent';
import Cart from '../Sales/Cart/Cart';

import './MyCourses.css';


const IndividualUsersCourseList = ({
operatorBusinessName,
operator,    
userId,
users, 
user,
courses,
coursesLoading,
onCoursesError }) => {


    if ( ! user ){

        navigate('/login');
    }


    if ( coursesLoading) {

        return <Loading />
    }         


   
    if ( onCoursesError ) {

        return <div> { onCoursesError.message } </div> ;
    }
    



    const viewAllCourses = () => {
        navigate(`/${operatorBusinessName}/courses`)
    }

    const viewTutorsBio = () => {
        navigate(`/${operatorBusinessName}/coursestaught/about/${userId}`)
    }
          

    let tutorsCourses, tutor, firstName;

    tutorsCourses = courses?.filter(course => course.createdBy === userId);

    tutor = users?.filter(thisUser => thisUser._id === userId)[0];

    firstName = user?.firstname;



    
    return (( tutorsCourses?.length === 0 ) ? (

       
        <div className="MyCourses">

            <header> 
                <h1>  {`Welcome ${firstName}! `} </h1>

                   
                    <div>  
                    <LoginLogout/>    

                    <Cart />
                    </div>
                </header>

        <br></br>   


    
             <div> 

             </div>

             <div>
                <h3>{tutor?.firstname} doesn't have any courses.  Please check back. <div>In the meantime, you can view other <h4><Link to={"/courses"}> courses. </Link></h4> </div></h3>
             </div>

         </div>

        ) : (

            <div className="MyCourses">

            <header> 
                <h1>  {`Welcome ${firstName}! `} </h1>
    
                <h2> You are viewing {tutor?.firstname}'s course list. </h2>

                <div className="lesson-item">  
                <LoginLogout/>

                <Cart />
                </div>
            </header>

            <br></br>   
            <button className="view-courses-btn" onClick={viewTutorsBio}>Bio</button>     
              <button className="view-courses-btn" onClick={viewAllCourses}>View All Courses</button> 

                    <CoursesComponent
                               operatorBusinessName={operatorBusinessName}
                               courses={tutorsCourses}
                   />     
            </div>
        )
    )     
}


 

const mapState = ( state, ownProps)  => ({
    user: state?.users?.user,
    users: getUsersByOperatorId(state, ownProps),
    yourCourses: getCoursesByOperatorId(state, ownProps)?.filter(course => state?.users.user?.courses?.includes(course?._id)),
    courses: getCoursesByOperatorId(state, ownProps),
    coursesLoading: state?.courses?.coursesLoading,
    onCoursesError: state?.courses?.onCoursesError
})

export default connect(mapState)(IndividualUsersCourseList);