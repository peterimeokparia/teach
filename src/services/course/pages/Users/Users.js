import React  from 'react';

import { 
connect } from 'react-redux';

import { 
navigate } from '@reach/router';

import Loading from '../Components/Loading';

import LoginLogout from '../Login/LoginLogout';

import MainMenu from '../Components/MainMenu';

import { 
navContent } from  '../Components/navigationHelper.js';

import { 
getOperatorFromOperatorBusinessName, 
getUsersByOperatorId,
getCoursesByOperatorId } from '../../Selectors';

import './CoursesComponent.css';


//shadow-panel

const Users = ({
operatorBusinessName,
operator,     
coursesLoading,
onCoursesError,
user,
users}) => {


    if ( ! user ){

        navigate(`/${operatorBusinessName}/login`);
    }


    if ( coursesLoading) {

        return <Loading />
    }         


   
    if ( onCoursesError ) {

        return <div> { onCoursesError.message } </div> ;
    }
    

    const viewCurrentUsersCourseList = ( userId ) => {
        navigate(`/${operatorBusinessName}/coursestaught/${userId}`); 
    }


    const gotToLessonPlan = ( user ) => { 

          navigate(`/${operatorBusinessName}/classroom/${user._id}`);
        //navigate(`/${operatorBusinessName}/LessonPlan/${user._id}/${user.firstname}`);
    }
               

    let navigationContent = navContent( user, operatorBusinessName, user?.role,  "Student" ).users;

    return (

            <div className="MyCourses">
       
               <header> 

               <MainMenu 
                    navContent={navigationContent}
               /> 
          
                 <h1>  {`Welcome ${user?.firstname}! `} </h1>

                   <LoginLogout
                        user={user}
                   />
                          
               </header>

                <div className="ComponentCourseList">

                <ul >
                    
                   {
                       users.map(singleUser => 
                       
                           <li className={"component-seconday-list-body"}> 
                             <div className={ "user-list-items"}>

                              {singleUser?.firstname}

                              <span className=""> <h6>{singleUser?.courses?.length} {singleUser?.courses?.length === 1  ? "Course.": "Courses."}  </h6></span>  
                               <button
                                        className="user-course-btn"
                                        onClick={() => viewCurrentUsersCourseList(singleUser?._id)}
                                        disabled={singleUser?.courses?.length === 0} 
                                >                                
                                   <span> {singleUser?.courses?.length === 1  ? "View Course." : (singleUser?.courses?.length === 0) ? "" : "View Courses."}  </span>
                                        
                                 </button>
                                 <button
                                        className="user-course-btn"
                                        onClick={() => gotToLessonPlan(singleUser)}
                                        disabled={singleUser?.courses?.length === 0} 
                                >                                
                                   <span> {"Go to Class Room."}  </span>
                                        
                                 </button>
                                </div> 
                           </li>
                           
                       )
                   }
                </ul>
              
            </div>

            </div>
        )
       
}




const mapState = (state, ownProps) => ({
    user: state?.users?.user,
    operator: getOperatorFromOperatorBusinessName(state, ownProps),
    users: getUsersByOperatorId(state, ownProps)?.filter(user => user?.role === "Tutor"),
    courses: getCoursesByOperatorId(state, ownProps),
    coursesLoading: state?.courses?.coursesLoading,
    onCoursesError: state?.courses?.onCoursesError
    
})

export default connect(mapState)(Users);