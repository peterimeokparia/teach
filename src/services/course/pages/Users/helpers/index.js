import { 
navigate } from '@reach/router';

import {
getCalendarColor, eventEnum } from 'services/course/pages/CalendarPage/helpers';

import { 
role } from 'services/course/helpers/PageHelpers';

import {
v4 as uuidv4 } from 'uuid';
  
import { 
formTypes } from 'services/course/pages/FormBuilder/helpers';

import { 
elementMeta } from "services/course/pages/QuestionsPage/helpers";

import Calendar from 'services/course/helpers/Calendar';
import Swal from 'sweetalert2';

export const links = (selectedStudents, courseId) => {
    return [ 
      { id: "SavedAnswers", title: "Saved Answers", path:`student/${ selectedStudents?._id }/savedanswers`, _id: selectedStudents?._id }, 
      { id: "Grades", title: "Grades", path:`student/${ selectedStudents?._id }/grades`, _id: selectedStudents?._id }, 
      { id: "Attendance", title: "Attendance", path: `student/${ selectedStudents?._id  }/attendance`, _id: selectedStudents?._id }, 
      { id: "Session", title: "Session", path: `student/${ selectedStudents?._id  }/sessions/courseId/${courseId}`, _id: selectedStudents?._id },
      { id: "Logins", title: "Logins", path:`student/${ selectedStudents?._id }/logins`, _id: selectedStudents?._id }, 
    ];
};

export const goToOnlineSurvey = ( operatorBusinessName, user ) => {
  let uuid = generateUuid();

  let userId = user?._id, 
      formUuId = uuid,
      formName = `dailysurvey_${uuid}`,
      formType = formTypes?.report

  navigate(`/${operatorBusinessName}/${formType}/${formName}/${formUuId}/${userId}`);
};

export const goToForms = ( formProps ) => {

  let {
    operatorBusinessName, 
    formName, 
    currentEventId,
    calendarEventType, 
    currentUser,
  } = formProps;

  const existingFormInTakingState =  takeExistingFormBuilderForm( formProps );

  if ( existingFormInTakingState && (existingFormInTakingState?.formType === formTypes.quizzwithpoints || existingFormInTakingState?.formType === formTypes.homework ) ) {

    navigate(`/${operatorBusinessName}/formBuilder/${existingFormInTakingState?.formType}/${existingFormInTakingState?.formName}/${existingFormInTakingState?.courseId}/${existingFormInTakingState?.lessonId}/${existingFormInTakingState?.formUuId}/${currentUser?._id}/${elementMeta.state.Taking}/${currentEventId}`);

  }

  if ( existingFormInTakingState && (existingFormInTakingState?.formType === formTypes.examwithpoints  )  ) {

    navigate(`/${operatorBusinessName}/formBuilder/${existingFormInTakingState?.formType}/${existingFormInTakingState?.formName}/${existingFormInTakingState?.courseId}/${existingFormInTakingState?.formUuId}/${currentUser?._id}/${elementMeta.state.Taking}/${currentEventId}`);
   
  }

  if ( existingFormInTakingState && (existingFormInTakingState?.formType === formTypes.report  )  ) {

    navigate(`/${operatorBusinessName}/formEventBuilder/${existingFormInTakingState?.formType}/${existingFormInTakingState?.formName}/${existingFormInTakingState?.formUuId}/${currentUser?._id}/${elementMeta.state.Taking}/${currentEventId}`);

  }

};

export function addUsersToMeeting( meetingProps, enableTeachPlatform ) {

  let {
    setAddUsers,
    listOfStudents, 
    selectedTutorId: selectedUserId, 
    operatorBusinessName, 
    sessions, 
    operator
  } = meetingProps;

  Swal.fire({
    title: 'Please Add Meeting Attendees',
    icon: 'warning',
    showCancelButton: true,
    confirmButtonText: 'Add Attendees',
    confirmButtonColor: '#673ab7',
    cancelButtonText: 'No Attendees'
  }).then( (response) => {
      if ( response?.value ) {       
          setAddUsers( true )
      } else {
        enableTeachPlatform({ listOfStudents, selectedTutorId: selectedUserId, operatorBusinessName, sessions, operatorId: operator?._id } );
        navigate(`/${operatorBusinessName}/LessonPlan/classRoom/${selectedUserId}`);
      }
  });
};

export const studentOption = ( students ) => students?.map(item => ( { value: item,  label: item?.firstname } ));

export function goToCalendar( props, user, eventType ) {

  let {
    users,
    calendars,
    calendar,
    operator,
    addCalendar,
    operatorBusinessName,
    courseId,
    lessonId,
    classRoomId,
    selectedCourse,
    lessonsByCourseId
  } = props;

  let currentCalendar = calendars?.find( cal => cal?.calendarEventType === eventType )

  if ( !currentCalendar ) {

      addCalendar( { calendar: new Calendar( calendarConfig( user, eventType, props ) )
        .calendar()})
        .then(calendar => {
        
        if ( calendar ) {

          navigateToCalendar( props, eventType, calendar, user );
          
        }
      })
      .catch( error => { console.log( error ); alert( JSON.stringify( error ) )} );

      return;
  } 

    navigateToCalendar( props, eventType, currentCalendar, user );

  return;
};

export function goToTimeLine( operatorBusinessName, eventType, user ){
  navigate(`/${operatorBusinessName}/schedule/${eventType}/timeline/${user._id}`);
};

export function viewCurrentUsersCourseList( operatorBusinessName, userId ){
  navigate(`/${operatorBusinessName}/coursestaught/${userId}`); 
};

export function gotToLessonPlan( operatorBusinessName, user ){ 
  navigate(`/${operatorBusinessName}/classroom/${user._id}`);
};

export function getCoursesSubscribedTo( tutor ){
  return tutor?.courses;
};

export function goToMeeting( goToMeetingProps, user ){
    
  let {
    enableTeachPlatform,
    usersAttendingMeeting,
    operatorBusinessName,
    sessions, 
    operator 
  } = goToMeetingProps;

  enableTeachPlatform({ listOfStudents: usersAttendingMeeting, selectedTutorId: user._id , operatorBusinessName, sessions, operator } )
  navigate(`/${operatorBusinessName}/LessonPlan/classRoom/${user._id}`);
};

export function getCoursesCreatedByUser( courses, tutor ){
  return courses?.
      filter( course => course?.createdBy === tutor?._id );
};

export function addStudentsToMeeting( addStudentsToMeetingProps, selectedUser, currentUser ){

  let {
    enableTeachPlatform,
    setSelectedUser,
    setAddUsers,
    listOfStudents, 
    operatorBusinessName, 
    sessions, 
    operator
  } = addStudentsToMeetingProps;

  setSelectedUser( selectedUser );

  if ( currentUser?.role === role.Tutor ) {

      let meetingProps = {
        setAddUsers,
        listOfStudents, 
        selectedTutorId: selectedUser._id , 
        operatorBusinessName, 
        sessions, 
        operator
      };

      if ( addUsersToMeeting( { ...meetingProps, selectedTutorId: selectedUser._id }, enableTeachPlatform ) ) {
          enableTeachPlatform({ listOfStudents: [], selectedTutorId: selectedUser._id, operatorBusinessName, sessions, operator } );
          navigate(`/${operatorBusinessName}/LessonPlan/classRoom/${selectedUser._id}`);
      }     

  } else {

      navigate(`/${operatorBusinessName}/LessonPlan/classRoom/${selectedUser._id}`);
  }

};

export function generateUuid(){
  const uuid = uuidv4();
  return uuid;
};

function calendarConfig( tutor, calendarType, props  ){


  let {
    users,
    calendars,
    operator,
    courseId,
    lessonId,
    classRoomId
  } = props;

  return {
    courseId,
    lessonId,
    classRoomId,
    users,
    userId: tutor?._id,
    calendarEventType: calendarType,
    operatorId: operator?._id,
    firstName: tutor?.firstname,
    color: getCalendarColor( calendars )
  }
};

function navigateToCalendar( props, calendarEventType, calendar, user ){ 

  let {
    operatorBusinessName,
    courseId,
    lessonId,
    selectedCourse,
  } = props;

  if ( calendarEventType === eventEnum.Lessons ) {
  
    navigate(`/${operatorBusinessName}/schedule/${calendarEventType}/calendar/${calendar?._id}/course/${courseId}/lesson/${lessonId}`);

    return;
  }

  navigate(`/${operatorBusinessName}/schedule/${calendarEventType}/calendar/${calendar?._id}/user/${user?._id}`);
};

function getFormType(calendarEventType){
  switch (calendarEventType) {
    case eventEnum.ReportForms:
      return formTypes.report
    case eventEnum.QuizzForms:
      return formTypes.quizzwithpoints
    default:
      break;
  }
}

function takeExistingFormBuilderForm( formProps ){ 

    let {
      operatorBusinessName, 
      currentUser, 
      events, 
      currentEventId,
      calendarEventType, 
      addNewFormBuilder,
      formBuilders
    } = formProps;

    const currentEventObject = events?.find( event => event?._id === currentEventId );
    const selectedFormBuilderObject = currentEventObject?.schedulingData[0];
    const selectedFormBuilder = formBuilders?.find( formBuilder => formBuilder?.formName === selectedFormBuilderObject?.formName && 
      formBuilder?.userId === currentUser?._id && 
      formBuilder?.eventId === currentEventId && 
      formBuilder?.state === elementMeta.state.Taking );

    if ( selectedFormBuilder && selectedFormBuilder?.state === elementMeta.state.Taking ) { 

      return selectedFormBuilder;

    }
 
    const formName = selectedFormBuilderObject?.formName;
    const formDisplayName = selectedFormBuilderObject?.formDisplayName;
    const formUuId = selectedFormBuilderObject?.formUuId;
    const formId = selectedFormBuilderObject?.formId;
    const courseId = selectedFormBuilderObject?.courseId;
    const lessonId = selectedFormBuilderObject?.lessonId;
    const createdBy = selectedFormBuilderObject?.createdBy;
    const createDateTime = selectedFormBuilderObject?.createDateTime;
    const eventId = currentEventId;
    const status = selectedFormBuilderObject?.status;
    const state = selectedFormBuilderObject?.state;
    
    let newBuilder = {
      operatorBusinessName: operatorBusinessName,
      formType: getFormType(calendarEventType),
      formDisplayName,
      formName,
      courseId,
      lessonId,
      formUuId: generateUuid(),
      formId,
      createDateTime,
      takingDateTime: Date.now(),
      createdBy,
      userId: currentUser?._id,
      status: elementMeta.status.InProgress,
      state: elementMeta.state.Taking,
      eventId
    };

   addNewFormBuilder( newBuilder );

  return true;  
};