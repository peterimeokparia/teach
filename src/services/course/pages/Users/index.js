import {
useState } from 'react';

import { 
connect } from 'react-redux';

import { 
navigate } from '@reach/router';

import {
navContent } from  'services/course/pages/components/NavigationHelper';

import { 
addCalendar } from 'services/course/actions/calendar';

import {
eventEnum,
getCalendarColor } from 'services/course/pages/CalendarPage/helpers';

import { 
getPushNotificationUsersByOperatorId,
getCalendarsByOperatorId,
getCalendarByCalendarEventType,
getEventsByOperatorId,    
getOperatorFromOperatorBusinessName, 
getUsersByOperatorId,
getCoursesByOperatorId } from 'services/course/selectors';

import {
permission,
SiteFunctionalityGroup,
Organization }from 'services/course/pages/components/SiteFunctionalityGroup';   // use this for permissions

import{
enableTeachPlatform } from 'services/course/actions/classrooms';

import {
studentOption,
goToCalendar,
goToTimeLine,
viewCurrentUsersCourseList,
gotToLessonPlan,
getCoursesSubscribedTo,
getCoursesCreatedByUser,
goToMeeting,
goToOnlineSurvey,
addStudentsToMeeting } from 'services/course/pages/Users/helpers';

import {
PageObject,
group } from 'services/course/pages/Users/helpers/permissions';

import { 
role } from 'services/course/helpers/PageHelpers';

import NotFoundPage from 'services/course/pages/components/NotFoundPage';
import Loading from '../components/Loading';
import LoginLogout from '../LoginPage/components/LoginLogout';
import MainMenu from 'services/course/pages/components/MainMenu';
import NavLinks from 'services/course/pages/components/NavLinks';
import SchoolIcon from '@material-ui/icons/School';
import ScheduleIcon from '@material-ui/icons/Schedule';
import TimelineIcon from '@material-ui/icons/Timeline';
import BookIcon from '@material-ui/icons/Book';
import ForumIcon from '@material-ui/icons/Forum';
import VideoCallIcon from '@material-ui/icons/VideoCall';
import PollIcon from '@mui/icons-material/Poll';
import Select from 'react-select';
import Calendar from 'services/course/helpers/Calendar';
import CalendarTodayIcon from '@material-ui/icons/CalendarToday';
import TabPanel from 'services/course/pages/components/FloatingActionButtonZoom';
import './style.css';

const Users = ({
    enableTeachPlatform,
    allSessions,
    allStudents,
    operatorBusinessName,
    operator,     
    coursesLoading,
    onCoursesError,
    calendarEvents,
    calendar,
    calendars,
    events,
    addCalendar,
    pushNotificationSubscribers,
    user,
    users,
    courses }) => {


    if ( ! operator || ! operatorBusinessName  ) {
        return <NotFoundPage />;
    }

    if ( ! user?.userIsValidated ){
        navigate(`/${operatorBusinessName}/login`);
    }

    if ( coursesLoading) {
        return <Loading />;
    }

    if ( onCoursesError ) {
        return <div> { onCoursesError.message } </div> ;
    }

    const [ addUsers, setAddUsers ] = useState( false );
    const [ selectedUser, setSelectedUser ] = useState( undefined );
    const [ usersAttendingMeeting, setUsersAttendingMeeting ] = useState( [] );
    const sessions = allSessions?.filter( usersession => usersession?.operatorId === operator?._id); 

    const calendarProps = {
        users,
        calendars,
        calendar,
        addCalendar,
        operatorBusinessName,
        operator
    };

    const goToMeetingProps = {
        enableTeachPlatform,
        usersAttendingMeeting,
        operatorBusinessName, 
        sessions, 
        operator
    };

    const addStudentsToMeetingProps = {
        enableTeachPlatform,
        setSelectedUser,
        setAddUsers,
        listOfStudents:[], 
        operatorBusinessName, 
        sessions, 
        operator
    };

return (
    <div className="Users">
        <div className={operatorBusinessName}>
            <header> 
                <MainMenu 
                    navContent={navContent( user, operatorBusinessName, user?.role,  "Student", [Organization.Boomingllc] ).users}
                />   
                <h1>  {`Welcome ${user?.firstname}! `} </h1>
                <div id="logout">
                    <LoginLogout
                        id="LoginLogout"
                        operatorBusinessName={operatorBusinessName}
                        user 
                    /> 
                </div>                   
            </header>
        </div>
            <div className="ComponentUserList">
            { ( addUsers ) 
                ?  <label>
                    Select users to add to this meeting:
                    <Select
                        placeholder={ `Add Students` }
                        isMulti
                        value={ usersAttendingMeeting }
                        onChange={ setUsersAttendingMeeting }
                        options={ studentOption( Object.values( allStudents ) ) } 
                    />    
                    <div>
                        <button
                            className="addStudentsToMeeting" 
                            onClick={() => goToMeeting( goToMeetingProps, selectedUser )}>
                            { "Continue"}
                        </button>
                    </div>
                    </label>        
                : <ul className={"component-seconday-ul"}>
                {users.map(singleUser => 
                    <li className={`component-seconday-list-body-users${operatorBusinessName === Organization.Boomingllc ? '-boomingllc' : ''}`} data-cy={`li_${singleUser?.firstname}`} key={singleUser?._id}> 
                        <div className={ "user-list-items-users"}>
                            <div className="row">
                                <div className="col-1"> 
                                    <img src={singleUser?.avatarUrl} width="80" height="80" alt=''/>
                                </div>
                                <div className="col-10">
                                 <NavLinks to={`/${operatorBusinessName}/coursestaught/about/${singleUser?._id}`}> 
                                    <div>
                                        <span id={`multicolortext`} className="multicolortext" data-cy={`multicolortext_${singleUser?.firstname}`}>  {singleUser?.firstname} </span> 
                                        <SiteFunctionalityGroup group={ permission( group, operatorBusinessName, PageObject.Users_Course_Count )}>
                                            <span className="price" name={PageObject.Users_Course_Count}> <h6>{getCoursesCreatedByUser( courses, singleUser )?.length} {getCoursesCreatedByUser( courses, singleUser )?.length === 1  ? "Course.": "Courses"}  </h6></span>
                                            <span className="price" name={PageObject.Users_Course_Count}> <h6>{getCoursesSubscribedTo(singleUser)?.length} {getCoursesSubscribedTo(singleUser)?.length === 1  ? "Subscribed Course.": "Subscribed Courses."}  </h6></span>
                                        </SiteFunctionalityGroup> 
                                    </div> 
                                </NavLinks>  
                                <SiteFunctionalityGroup group={ permission( group, operatorBusinessName, PageObject.Users_VideoCallIcon )}> 
                                <VideoCallIcon 
                                    id="VideoCallIcon"
                                    name={PageObject.Users_VideoCallIcon}
                                    data-cy={`${(singleUser?.firstname)?.toLowerCase()}_VideoCallIcon`}
                                    className="round-button-3"
                                    onClick={() => addStudentsToMeeting( addStudentsToMeetingProps, singleUser, user )}
                                /> 
                                </SiteFunctionalityGroup>
                                <SiteFunctionalityGroup group={ permission( group, operatorBusinessName, PageObject.Users_SchoolIcon )}>
                                <SchoolIcon 
                                        id="SchoolIcon"
                                        name={PageObject.Users_SchoolIcon}
                                        data-cy={`${(singleUser?.firstname)?.toLowerCase()}_SchoolIcon`}
                                        className="round-button-1"
                                        onClick={() => viewCurrentUsersCourseList(operatorBusinessName, singleUser?._id)}
                                        disabled={singleUser?.courses?.length === 0} 
                                    />
                                </SiteFunctionalityGroup>
                                <SiteFunctionalityGroup group={ permission( group, operatorBusinessName, PageObject.Users_BookIcon )}>  
                                    <BookIcon 
                                        id="BookIcon"
                                        name={PageObject.Users_BookIcon}
                                        data-cy={`${(singleUser?.firstname)?.toLowerCase()}_BookIcon`}
                                        className="round-button-2"
                                        onClick={() => gotToLessonPlan( operatorBusinessName, singleUser)}
                                        disabled={singleUser?.courses?.length === 0} 
                                    />
                                    </SiteFunctionalityGroup>
                                    <SiteFunctionalityGroup group={ permission( group, operatorBusinessName, PageObject.Users_ScheduleIcon )}>
                                    <ScheduleIcon
                                        id="ScheduleIcon"
                                        name={PageObject.Users_ScheduleIcon}
                                        data-cy={`${(singleUser?.firstname)?.toLowerCase()}_ScheduleIcon`}
                                        className="round-button-1"
                                        onClick={() => goToCalendar( calendarProps, singleUser, eventEnum.NewEvent )}
                                        disabled={singleUser?.courses?.length === 0}  
                                    />
                                    </SiteFunctionalityGroup>
                                    {/*
                                    <SiteFunctionalityGroup group={ permission( group, operatorBusinessName, PageObject.Users_CalendarTodayIcon )}>
                                     <CalendarTodayIcon 
                                        id="CalendarTodayIcon"
                                        name={PageObject.Users_CalendarTodayIcon}
                                        data-cy={`${(singleUser?.firstname)?.toLowerCase()}_CalendarTodayIcon`}
                                        className="round-button-5"
                                        onClick={() => goToCalendar(calendarProps, singleUser, eventEnum?.ConsultationForm)}
                                        disabled={singleUser?.courses?.length === 0} 
                                    /> 
                                    </SiteFunctionalityGroup>
                                    */}
                                    <SiteFunctionalityGroup group={ permission( group, operatorBusinessName, PageObject.Users_ForumIcon )}>
                                    <ForumIcon 
                                        id="ForumIcon"
                                        name={PageObject.Users_ForumIcon}
                                        data-cy={`${(singleUser?.firstname)?.toLowerCase()}_ForumIcon`}
                                        className="round-button-6"
                                        onClick={() => goToCalendar(calendarProps, singleUser, eventEnum?.OnlineTutoringRequest)}
                                        disabled={singleUser?.courses?.length === 0} 
                                    />
                                    </SiteFunctionalityGroup>
                                    <SiteFunctionalityGroup group={ permission( group, operatorBusinessName, PageObject.Users_VideoCallIcon )}> 
                                    <PollIcon 
                                        id="PollIcon"
                                        name={PageObject.Users_PollIcon}
                                        data-cy={`${(singleUser?.firstname)?.toLowerCase()}_PollIcon`}
                                        className="round-button-3"
                                        onClick={() => goToCalendar( calendarProps, singleUser, eventEnum?.ReportForms )}
                                        // onClick={() => goToOnlineSurvey( operatorBusinessName, user )}
                                    /> 
                                    </SiteFunctionalityGroup>
                                    <SiteFunctionalityGroup group={ permission( group, operatorBusinessName, PageObject.Users_TimelineIcon )}>
                                    <TimelineIcon 
                                        id="TimelineIcon"
                                        name={PageObject.Users_TimelineIcon}
                                        data-cy={`${(singleUser?.firstname)?.toLowerCase()}_TimelineIcon`}
                                        className="round-button-7"
                                        onClick={() => goToTimeLine(operatorBusinessName, eventEnum.SessionScheduling, singleUser)}
                                        disabled={singleUser?.courses?.length === 0} 
                                    />
                                    </SiteFunctionalityGroup>
                                </div>
                            </div>
                        </div> 
                    </li>    
                    )
                }
            </ul>
            }
    </div>
    </div>
    );       
};

const mapState = ( state, ownProps ) => ({
    user: state?.users?.user,
    allStudents: getUsersByOperatorId(state, ownProps)?.filter(user => user?.role === role.Student),
    operator: getOperatorFromOperatorBusinessName(state, ownProps),
    calendar: getCalendarByCalendarEventType(state, ownProps),
    calendars: getCalendarsByOperatorId(state, ownProps),
    events: getEventsByOperatorId(state, ownProps),
    users: getUsersByOperatorId(state, ownProps)?.filter(user => user?.role === role.Tutor),
    courses: getCoursesByOperatorId(state, ownProps),
    coursesLoading: state?.courses?.coursesLoading,
    onCoursesError: state?.courses?.onCoursesError,
    pushNotificationSubscribers: getPushNotificationUsersByOperatorId(state, ownProps),
    allSessions: Object.values(state?.sessions?.sessions)
});

export default connect(mapState, { addCalendar, enableTeachPlatform })(Users);