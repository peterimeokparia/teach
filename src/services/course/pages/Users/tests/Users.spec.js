import React from 'react';

import { 
Provider, 
useSelector } from "react-redux";

import { 
applyMiddleware } from 'redux'; 

import { 
mockStoreObject } from 'services/course/api';

import Users  from 'services/course/pages/Users';
import renderer from 'react-test-renderer';
import thunk from "redux-thunk"; 
import * as reactRedux from 'react-redux';
import configureStore from 'redux-mock-store';

  const mockStore = configureStore([]);

describe('Users', () =>  {  
  let  store = mockStore(mockStoreObject, applyMiddleware(thunk));

  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
    jest.spyOn(Date.prototype, 'toLocaleTimeString')
        .mockImplementation(() => '12:00:00');
    jest.spyOn(Date.prototype, 'toLocaleDateString')
      .mockImplementation(() => 'AM 1/01/2021');
  });

  it('renders as expected', async () => {
    const dispatch = jest.fn();
    useDispatchMock.mockReturnValue(dispatch);
    const tree = renderer.create(
      <Provider store={store}>
        <Users /> 
      </Provider>
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});




