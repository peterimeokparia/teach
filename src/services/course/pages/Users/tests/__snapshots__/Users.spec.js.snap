// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users renders as expected 1`] = `
<div>
  \`Not Found. Fooled you! HAHAHA and you thought that I exist.\`
</div>
`;
